src/Bicep.Core/Emit/EmitLimitationCalculator.cs

```diff
@@ -446,5 +448,31 @@ private static bool IsInvariant(SemanticModel semanticModel, LocalVariableSyntax
                 ? IsLocalInvariant(itemVariable)
                 : IsLocalInvariant(itemVariable) && IsLocalInvariant(indexVariable);
         }
+
+        private static void BlockModuleOutputResourcePropertyAccess(SemanticModel model, IDiagnosticWriter diagnostics) =>
+            diagnostics.WriteMultiple(model.AllResources.OfType<ModuleOutputResourceMetadata>()
+                // find references to this resource
+                .SelectMany(moduleOutputResource => model.FindReferences(moduleOutputResource.Module)
```

Comment:
`FindReferences()` is quite inefficient as it was designed for use in ad-hoc language server handlers (find references, rename, etc) - it uses a visitor internally without caching, and as a result is O(syntax_nodes). I wouldn't recommend depending on it for this type of scenario.

Reply:
I'm wondering if another possible alternate approach would be to use the pre-calculated module output metadata directly, with one pass over the syntax tree - e.g. something like:

```csharp
var syntaxToBlock = SyntaxAggregator.Aggregate(model.Root.Syntax, syntax => 
    syntax is PropertyAccessSyntax propertyAccess &&
    model.ResourceMetadata.TryLookup(propertyAccess.BaseExpression) is ModuleOutputResourceMetadata resource &&
    !ResourcePropertiesResolvedAtCompileTime.Contains(propertyAccess.PropertyName.IdentifierName));
```

Reply:
Isn't this still O(syntax_nodes), since SyntaxAggregator will iterate over every leaf in the syntax tree? I could start from the binder and only look at nodes that are direct parents of syntax bound to a module symbol known to host a module output resource (though this does require keeping track of module output resources in an accessible way, which may have been deliberately skipped per https://github.com/Azure/bicep/pull/9646#discussion_r1087930367)

Reply:
Sorry, I should have explained that better - you're using an `O(syntax_nodes)` method inside a Linq expression which is being executed once per resource output - overall it's going to be `O(syntax_nodes * num_resource_outputs)` - it's the multiplicative factor there I'm objecting to.

Reply:
😮! I didn't check the implementation of FindReferences before using it and thought it would basically find the keys in model's bindings cache that had been mapped to a given symbol. Thanks for the heads up!